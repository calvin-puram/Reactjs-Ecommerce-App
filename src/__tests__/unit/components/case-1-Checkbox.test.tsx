import React from 'react'
import { axe } from 'jest-axe'
import { render, fireEvent } from '@testing-library/react'
import ReactDom from 'react-dom'
// import jest from 'jest'

import Checkbox from '../../../components/Checkbox'

/**
 * This checkbox component renders a checkbox with a label.
 * Since we customized the default checkbox, we want to
 * make sure it still works as a regular checkbox
 * should.
 */
describe('The <Checkbox /> component', () => {
  const CheckboxProps = {
    label: 'checkbox checker',
    id: 'checkbox',
    checked: false,
    background: '#000',
    checkMarkBackground: '#fff',
    onChange: jest.fn(),
  }
  const setupCheckBox = () => render(<Checkbox {...CheckboxProps} />)
  it(' Should render the label and checkbox the user will see', () => {
    const { asFragment } = setupCheckBox()
    expect(asFragment()).toMatchSnapshot()
  })

  it(' Should make the checkbox accessible by setting the id and htmlFor attributes on label and checkbox', () => {
    const { getByLabelText, debug } = setupCheckBox()
    expect(getByLabelText(CheckboxProps.label)).toBeInTheDocument()
  })

  it('❌ Should call the onChange handler when it is provided', () => {})

  it('❌ Should change state correctly when clicked (checked and unchecked)', () => {})

  it('❌ should not fail any accessibility tests', async () => {})
})
