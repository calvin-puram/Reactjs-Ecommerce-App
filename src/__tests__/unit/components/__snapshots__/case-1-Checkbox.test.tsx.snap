// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Checkbox /> component  Should render the label and checkbox the user will see 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx"
    data-testid="Checkbox"
  >
    <input
      class="sc-hKFyIo cqtmJr"
      id="checkbox"
      type="checkbox"
    />
    <label
      class="sc-dlnjPT iMZIHZ"
      for="checkbox"
    >
      <span
        class="sc-gtssRu SzCNg"
      >
        checkbox checker
      </span>
    </label>
  </div>
</DocumentFragment>
`;
